docs/use-cases/ai/mcp-server-enable-third-party-ai-agent-access.mdx

```diff
@@ -0,0 +1,95 @@
+
+import thirdPartyAppPermissions from './assets/third-party-app-permissions.png';
+import Prerequisites from './fragments/_mcp-prerequisites.mdx';
+import SampleCode from './fragments/_mcp-sample-code.mdx';
+import SetUpServer from './fragments/_mcp-set-up-server.mdx';
+
+# Enable third-party AI agent access to your MCP server
+
+:::note
+If you want to integrate your MCP server with your own AI agent or app, please refer to the [Enable auth for your MCP-powered apps with Logto](./mcp-server-add-auth) guide.
+:::
+
+This guide walks you through integrating Logto with your MCP server using [mcp-auth](https://mcp-auth.dev), allowing you to authenticate users and securely retrieve their identity information using the standard OpenID Connect flow.
+
+You'll learn how to:
+
+- Configure Logto as the authorization server for your MCP server.
+- Set up a “whoami” tool to return the current user's identity claims.
+- Test the flow with a third-party AI agent.
+
+After this tutorial, your MCP server will:
+
+- Authenticate users in your Logto tenant.
+- Return identity claims (`sub`, `username`, `name`, `email`, etc.) for the "whoami" tool invocation.
+
+<Prerequisites clientDescription="The third-party AI agent will be used as the client in this guide." />
+
+## Set up third-party AI agent
+
+To enable the third-party AI agent to access your MCP server, you need to set up the following:
+
+1. The client should be able to make MCP requests to invoke the tools exposed by the MCP server.
+2. The client should be able to handle the 401 Unauthorized response. See [Authorization Flow Steps](https://modelcontextprotocol.io/specification/2025-03-26/basic/authorization#2-5-authorization-flow-steps) for more details.
+3. After successful authentication, the client should be able to make requests to the MCP server with the access token obtained from Logto.
+
+## Set up AI agent in Logto
+
+To enable the third-party AI agent to access your MCP server, you need to set up a **third-party app** in Logto. This app will be used to represent the AI agent and obtain the necessary credentials for authentication and authorization.
+
+### Allow developers to create third-party apps in Logto
+
+If you are building a marketplace or want to allow developers to create third-party apps in Logto, you can leverage [Logto Management API](/integrate-logto/interact-with-management-api) to create third-party apps programmatically. This allows developers to register their applications and obtain the necessary credentials for authentication.
+
+You'll need to host your own service to handle the client registration process. This service will interact with the Logto Management API to create third-party apps on behalf of developers.
+
+```mermaid
+flowchart LR
+  A[Developer] -->|Register app| B[Your app registration service]
+  B -->|Create app| C[Logto Management API]
+  C -->|App ID, secret, etc.| B
+  B -->|App ID, secret, etc.| A
+```
```

Comment:
Better to use a sequence diagram, because with multiple entities interacting back and forth, the processing order of the flow isn’t very intuitive to grasp.